### ISSUE TITLE
On RoCM, some torch.compile fails with 0 mismatched elements

### ISSUE THREAD

**Issue by isuruf** [pytorch/pytorch#132283]
### 🐛 Describe the bug

The tests fail with
```
AssertionError: Tensor-likes are not equal!

Mismatched elements: 0 / 8000 (0.0%)
Greatest absolute difference: 0 at index (0,)
Greatest relative difference: 0.0 at index (0,)
```

Examples:
https://github.com/pytorch/pytorch/issues/131158
https://github.com/pytorch/pytorch/issues/131118
https://github.com/pytorch/pytorch/issues/131748
https://github.com/pytorch/pytorch/issues/131727
https://github.com/pytorch/pytorch/issues/131627

### Versions

See above

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @ezyang @chauhang @penguinwu

**Comment by jeffdaily** [comment]
This is expected to be fixed in ROCm 6.2.1 after upgrading our CI servers to that ROCm version.

**Comment by jeffdaily** [comment]
Our CI servers were upgraded to ROCm 6.2.1.  Please reopen this ticket if you continue to see errors like this.
